src/ecalc/libraries/libecalc/common/libecalc/common/dump.py

```diff
@@ -0,0 +1,56 @@
+from functools import wraps
+from pathlib import Path
+from typing import Optional, Any, Callable
+
+from libecalc.common.logger import logger
+
+unique_id: int = 1
+"""
+global static id to increase by one for each file being dumped
+"""
+
+
+def dump(file_dump: Path, captured: Any):
+    """
+    A simple dump function to dump the content that must be serializable. Each time this function is
+    called the name of the file that data is dumped to (which is an integer) will be incremented with 1.
+
+    :param file_dump:   Where to dump it. Missing parent directories will be created.
+    :param captured:    What to dump.
+    """
+    Path.mkdir(file_dump, parents=True, exist_ok=True)
+    global unique_id
+    file_dump_name = file_dump / f"{str(unique_id)}.json"
+    unique_id += 1
+
+    with open(file_dump_name, 'a+') as file:
+        file.write(str(captured))
+
+
+class Capture:
+    """Decorator to capture return values from func and dump to specified file"""
+
+    @staticmethod
+    def dump(dump_return_values: bool = False, file_dump: Path = "dump") -> Optional[Any]:
```

Comment:
Confusing with two 'dump' functions?

I understood 'dump_return_values = False' as only dump input data. But it does not dump anything when false?

Reply:
I see now that there are more possibilities yes, I will change a bit